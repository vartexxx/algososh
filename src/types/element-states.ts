export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified",
}

export type ElementTypes = {
  letter: string;
  state: ElementStates;
};

export type SortTypes = {
  index: number;
  state: ElementStates;
};

export enum CirclePosition {
  head = "head",
  tail = "tail",
}

export type CircleState = {
  modifiedIndex: number;
  changingIndex: number;
};

