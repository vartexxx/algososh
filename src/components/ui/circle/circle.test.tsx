import renderer, {ReactTestRendererJSON} from "react-test-renderer";
import {Circle} from "./circle";
import {ElementStates} from "../../../types/element-states";


describe('Тестирование компонента Circle', (): void => {
    test('Компонент Circle корректно рендерится без буквы', (): void => {
        const letter: string = '';
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Circle letter={letter} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Компонент Circle корректно рендерится с буквами', (): void => {
        const letter: string = 'Буквы';
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Circle letter={letter} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Компонент Circle корректно рендерится с head', (): void => {
        const head: string = 'head';
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Circle head={head} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Компонент Circle корректно рендерится с react-элементом в head', (): void => {
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Circle head={<Circle letter={'letter'} />} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Компонент Circle корректно рендерится с tail', (): void => {
        const tail: string = 'tail';
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Circle tail={tail} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Компонент Circle корректно рендерится с react-элементом в tail', (): void => {
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Circle tail={<Circle letter={'letter'} />} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Компонент Circle корректно рендерится с index', (): void => {
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Circle index={0} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Компонент Circle корректно рендерится с пропсом isSmall == true', (): void => {
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Circle isSmall={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Компонент Circle корректно рендерится в состоянии default', (): void => {
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null  = renderer.create(<Circle state={ElementStates.Default} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('Компонент Circle корректно рендерится в состоянии changing', (): void => {
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null  = renderer.create(<Circle state={ElementStates.Changing} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('Компонент Circle корректно рендерится в состоянии modified', (): void => {
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null  = renderer.create(<Circle state={ElementStates.Modified} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
})