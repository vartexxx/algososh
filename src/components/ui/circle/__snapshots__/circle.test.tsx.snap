// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Circle Компонент Circle корректно рендерится без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Компонент Circle корректно рендерится в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Компонент Circle корректно рендерится в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Компонент Circle корректно рендерится в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Компонент Circle корректно рендерится с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Компонент Circle корректно рендерится с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Компонент Circle корректно рендерится с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          letter
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Компонент Circle корректно рендерится с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          letter
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Тестирование компонента Circle Компонент Circle корректно рендерится с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Тестирование компонента Circle Компонент Circle корректно рендерится с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Буквы
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Компонент Circle корректно рендерится с пропсом isSmall == true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
